virtme/commands/run.py

```diff
@@ -858,6 +868,18 @@ def can_use_kvm(args):
     return can_access_file("/dev/kvm")
 
 
+def all_tools_available(tools: List[str]) -> bool:
+    return all(map(lambda tool: which(tool) is not None, tools))
+
+
+def can_use_ssh_over_vsock(args) -> bool:
+    return (
+        not args.force_tcp
+        and all_tools_available(["socat", "setsid", "systemd-socket-activate"])
+        and can_access_file("/dev/vhost-vsock")
```

Comment:
Similar to your other issue inside the VM: is this "file" always present, or does the corresponding kernel module need to be loaded first?

Reply:
```
Vhost-vsock is a software device so there is no probe call that causes
the driver to register its misc char device node.  This creates a
chicken and egg problem: userspace applications must open
/dev/vhost-vsock to use the driver but the file doesn't exist until the
kernel module has been loaded.

Use the devname modalias mechanism so that /dev/vhost-vsock is created
at boot.  The vhost_vsock kernel module is automatically loaded when the
first application opens /dev/host-vsock.
``` [1]

So seems like if the device is opened, the kernel module is getting loaded.

[1] https://patchwork.kernel.org/project/kvm/patch/20170509192858.17623-1-stefanha@redhat.com/